Allow remote service creds to be configured using env (#30797)
hashicorp/terraform: website/docs/cli/config/config-file.mdx

Terraform Cloud responds to API calls at both its current hostname
`app.terraform.io`, and its historical hostname `atlas.hashicorp.com`.

### Environment Variable Credentials

If you would prefer not to store your API tokens directly in the CLI
configuration, you may use a host-specific environment variable. For example,
the value of a variable named `TF_TOKEN_app_terraform_io` will be used as an
Authorization: Bearer token whenever making service requests to the hostname
"app.terraform.io".

> Internationalized domain names should be converted to their [punycode equivalent](https://www.charset.org/punycode)
> with an ACE prefix when used as a variable name. For example, token credentials
> for 例えば.com should be set in a variable called `TF_TOKEN_xn--r8j3dr99h_com`.

Reviewer: ```suggestion
Convert domain names to their [punycode equivalent](https://www.charset.org/punycode) and include an ACE prefix when using them as a variable name. For example, set the token credentials for 例えば.com in a variable called `TF_TOKEN_xn--r8j3dr99h_com`.
```

Author: Thanks for your suggestions! I think something was lost in this modification, because this only applies to domain names containing non-ASCII characters, typical of domain names with accent marks or unicode characters for non-latin character sets. Should it say...

> Internationalized domain names containing non-ASCII chacters should be converted

?

Reviewer: Is "internationalized" a common term? If not, I think "Convert domain names with non-ASCII characters to...."  

Thank you for pushing back on that!

Reviewer 2: "[Internationalized domain name](https://en.wikipedia.org/wiki/Internationalized_domain_name)" as a whole is the particular jargon phrase used to refer to the family of specifications that this "punycode" formatting is a part of. Since we're already linking to a reference for "punycode" anyway we may not need to _also_ say "Internationalized domain name", but if we do then I'd suggest linking to a definition for it so it's clearer it's not just a regular adjective modifying "domain name".

Terraform Cloud responds to API calls at both its current hostname
`app.terraform.io`, and its historical hostname `atlas.hashicorp.com`.

### Environment Variable Credentials

If you would prefer not to store your API tokens directly in the CLI
configuration, you may use a host-specific environment variable. Environment variable names should
have the prefix `TF_TOKEN_` added to the domain name, with periods encoded as underscores.
For example, the value of a variable named `TF_TOKEN_app_terraform_io` will be used as a
bearer authorization token when the CLI makes service requests to the hostname "app.terraform.io".

> Domain names containing non-ASCII characters should be converted to
> their [punycode equivalent](https://www.charset.org/punycode)
> with an ACE prefix when used as a variable name. For example, token credentials
> for 例えば.com must be set in a variable called `TF_TOKEN_xn--r8j3dr99h_com`.

Reviewer: Just working on flow and making sure we're using active voice :-) 

```suggestion
When using domain names as a variable name, you must convert domain names containing non-ASCII characters to their [punycode equivalent](https://www.charset.org/punycode) with an ACE prefix. For example, token credentials for 例えば.com must be set in a variable called `TF_TOKEN_xn--r8j3dr99h_com`.
```

> Hyphens are allowed within variable names by certain tools, such as the `env` utility,
> but are not valid POSIX variable names. Therefore, it can be tricky to
> set them with interactive tools such as bash or zsh so you may encode
> hyphens as double underscores. For example, a token for the domain name "café.fr" can
> be set by either `TF_TOKEN_xn--caf-dma_fr` or `TF_TOKEN_xn____caf__dma_fr` if that is more
> convenient. If both are defined, the variable containing hyphens is preferred.

Reviewer: I'm working on active voice and also just making things clearer by fussing with the way information is presented in the sentences. As usual - please push back on me if I've edited in a way that loses/changes the meaning :-) Thank you so much for working with me on this! I appreciate you adding in these extra details for users.

```suggestion
Some tools like the `env` utility allow hyphens in variable names, but hyphens create invalid POSIX variable names. Therefore, you can encode hyphens as double underscores when you set variables with interactive tools like Bash or Zsh. For example, you can set a token for the domain name "café.fr" as either `TF_TOKEN_xn--caf-dma_fr` or `TF_TOKEN_xn____caf__dma_fr`. If both are defined, Terraform will use the version containing hyphens.
```
